### ISSUE TITLE
Several Build Tasks with failure with Agent V2

### ISSUE THREAD

**Issue by PArpagaus** [Microsoft/vsts-tasks#4812]
In connection with the migration to Visual Studio 2017 it was necessary to update the build agents on the Build Server (OnPremise) from version 1.x to 2.112.0. Unfortunately, a lot of the build tasks aren’t functioning correctly now. Especially all “NuGet” and the “Windows Machine File Copy” task are failing, so that it isn’t possible to create and publish Nuget packages automatically for example. I can’t find any alternative tasks or an update to the existing tasks to solve this problem.

The following error message will be displayed in the log of the build definition:

2017-07-10T11:17:30.9134615Z Executing the powershell script: C:\Builds\Agent_1\_tasks\NuGetPackager_333b11bd-d341-40d9-afcf-b32d5ce6f24b\0.1.72\NuGetPackager.ps1 
2017-07-10T11:17:31.5072276Z ##[error]System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.TeamFoundation.DistributedTask.Agent.Interfaces, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified. 
2017-07-10T11:17:31.5072276Z File name: 'Microsoft.TeamFoundation.DistributedTask.Agent.Interfaces, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' 
2017-07-10T11:17:31.5072276Z at System.Reflection.RuntimeAssembly.GetExportedTypes(RuntimeAssembly assembly, ObjectHandleOnStack retTypes) 
2017-07-10T11:17:31.5072276Z at System.Reflection.RuntimeAssembly.GetExportedTypes() 
2017-07-10T11:17:31.5072276Z at System.Management.Automation.Runspaces.PSSnapInHelpers.GetAssemblyTypes(Assembly assembly, String name) 
2017-07-10T11:17:31.5072276Z at System.Management.Automation.Runspaces.PSSnapInHelpers.AnalyzePSSnapInAssembly(Assembly assembly, String name, PSSnapInInfo psSnapInInfo, PSModuleInfo moduleInfo, Boolean isModuleLoad, Dictionary`2& cmdlets, Dictionary`2& providers, String& helpFile) 
2017-07-10T11:17:31.5072276Z at System.Management.Automation.Runspaces.InitialSessionState.ImportCmdletsFromAssembly(Assembly assembly, PSModuleInfo module) 
2017-07-10T11:17:31.5072276Z at Microsoft.PowerShell.Commands.ModuleCmdletBase.LoadBinaryModule(PSModuleInfo parentModule, Boolean trySnapInName, String moduleName, String fileName, Assembly assemblyToLoad, String moduleBase, SessionState ss, ImportModuleOptions options, ManifestProcessingFlags manifestProcessingFlags, String prefix, Boolean loadTypes, Boolean loadFormats, Boolean& found, String shortModuleName, Boolean disableFormatUpdates) 
2017-07-10T11:17:31.5072276Z at Microsoft.PowerShell.Commands.ModuleCmdletBase.LoadModule(PSModuleInfo parentModule, String fileName, String moduleBase, String prefix, SessionState ss, Object privateData, ImportModuleOptions& options, ManifestProcessingFlags manifestProcessingFlags, Boolean& found, Boolean& moduleFileFound) 
2017-07-10T11:17:31.5072276Z at Microsoft.PowerShell.Commands.ModuleCmdletBase.LoadUsingExtensions(PSModuleInfo parentModule, String moduleName, String fileBaseName, String extension, String moduleBase, String prefix, SessionState ss, ImportModuleOptions options, ManifestProcessingFlags manifestProcessingFlags, Boolean& found, Boolean& moduleFileFound) 
2017-07-10T11:17:31.5072276Z at Microsoft.PowerShell.Commands.ModuleCmdletBase.LoadUsingModulePath(PSModuleInfo parentModule, Boolean found, IEnumerable`1 modulePath, String name, SessionState ss, ImportModuleOptions options, ManifestProcessingFlags manifestProcessingFlags, PSModuleInfo& module) 
2017-07-10T11:17:31.5072276Z at Microsoft.PowerShell.Commands.ImportModuleCommand.ImportModule_LocallyViaName(ImportModuleOptions importModuleOptions, String name) 
2017-07-10T11:17:31.5072276Z at Microsoft.PowerShell.Commands.ImportModuleCommand.ProcessRecord() 
2017-07-10T11:17:31.5072276Z at System.Management.Automation.CommandProcessor.ProcessRecord()

**Comment by ericsciple** [comment]
Are you on VSTS or are you using on-premises TFS? If TFS, which version and update?

Also, I would recommend creating a separate issue for the WindowsMachineFileCopy task since it is not owned by the same team that owns the NuGet tasks.

**Comment by PArpagaus** [comment]
I am using on-premises TFS 2017 SP1, Product Version 15.112.26307.0. The problem with all the tasks is the same, they all imports the following modules:

import-module "Microsoft.TeamFoundation.DistributedTask.Task.Internal"
import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"

These modules are no longer part of the new build agent.

**Comment by keithrob** [comment]
@PArpagaus,
Are you still having this issue, and if so can you let me know which version of the Nuget build task you are using?

In fact, running the build again with [debug enabled](https://www.visualstudio.com/en-us/docs/build/define/variables#control-variables) and attaching those logs would be helpful.

Keith

**Comment by PArpagaus** [comment]
Hello 

 

If have still the same issue, temporary I copied the necessary assemblies to the GAC to let work the tasks. The version is as follow:

 

2017-09-05T20:01:37.7746575Z ==============================================================================

2017-09-05T20:01:37.7746575Z Task         : NuGet Packager

2017-09-05T20:01:37.7746575Z Description  : Creates nupkg outputs from csproj or nuspec files

2017-09-05T20:01:37.7746575Z Version      : 0.1.72

2017-09-05T20:01:37.7746575Z Author       : Lawrence Gripper

2017-09-05T20:01:37.7746575Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=627416)

2017-09-05T20:01:37.7746575Z ==============================================================================

 

2017-09-05T20:02:51.1037340Z ==============================================================================

2017-09-05T20:02:51.1037340Z Task         : NuGet Publisher

2017-09-05T20:02:51.1037340Z Description  : Uploads nupkg files to a NuGet server

2017-09-05T20:02:51.1037340Z Version      : 0.2.30

2017-09-05T20:02:51.1037340Z Author       : Lawrence Gripper

2017-09-05T20:02:51.1037340Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=627417)

2017-09-05T20:02:51.1037340Z ==============================================================================

I will upload a debug enabled log as soon as possible.

 

Kind regards

Patrick

 

 

Von: Keith Robertson [mailto:notifications@github.com] 
Gesendet: Dienstag, 22. August 2017 17:53
An: Microsoft/vsts-tasks <vsts-tasks@noreply.github.com>
Cc: Patrick Arpagaus <patrick.arpagaus@weroSoft.net>; Mention <mention@noreply.github.com>
Betreff: Re: [Microsoft/vsts-tasks] Several Build Tasks with failure with Agent V2 (#4812)

 

@PArpagaus <https://github.com/parpagaus> ,
Are you still having this issue, and if so can you let me know which version of the Nuget build task you are using?

In fact, running the build again with debug enabled <https://www.visualstudio.com/en-us/docs/build/define/variables#control-variables>  and attaching those logs would be helpful.

Keith

**Comment by keithrob** [comment]
Still waiting on logs.  Closing this out.
